pkg/etcdutil/defrag_test.go

```diff
@@ -0,0 +1,59 @@
+package etcdutil_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/gardener/etcd-backup-restore/pkg/etcdutil"
+)
+
+var _ = Describe("Defrag", func() {
+	var (
+		tlsConfig             *TLSConfig
+		endpoints             = []string{"http://localhost:2379"}
+		etcdConnectionTimeout = time.Duration(30 * time.Second)
+		keyPrefix             = "/defrag/key-"
+		valuePrefix           = "val"
+	)
+	tlsConfig = NewTLSConfig("", "", "", true, true, endpoints)
+	Context("Defragmentation", func() {
+		BeforeEach(func() {
+			client, err := GetTLSClientForEtcd(tlsConfig)
+			defer client.Close()
+			Expect(err).ShouldNot(HaveOccurred())
+			for index := 0; index <= 100; index++ {
+				ctx, cancel := context.WithTimeout(context.TODO(), etcdConnectionTimeout)
+				client.Put(ctx, fmt.Sprintf("%s%d", keyPrefix, index), valuePrefix)
+				cancel()
+			}
+			for index := 0; index <= 50; index++ {
+				ctx, cancel := context.WithTimeout(context.TODO(), etcdConnectionTimeout)
+				client.Delete(ctx, fmt.Sprintf("%s%d", keyPrefix, index))
+				cancel()
+			}
+		})
+		It("should defragment and reduce size of DB", func() {
```

Comment:
Maybe also add a negative test case? Say, configure timeout in such a way the it would be guaranteed to timeout? Or nothing to defrag?